api/pkg/client/mlflow/mlflow_test.go

```diff
@@ -0,0 +1,348 @@
+		t.Run(tc.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+				w.WriteHeader(http.StatusOK)
+				_, err := w.Write([]byte(tc.expectedRespJSON))
+				require.NoError(t, err)
+			}))
+			defer server.Close()
+			client := NewMlflowClient(server.Client(), Config{
+				TrackingURL: server.URL,
+			}, &mocks.GcsService{})
+
+			resp, errAPI := client.SearchRunForExperiment(tc.idExperiment)
+			assert.Equal(t, tc.expectedError, errAPI)
+			assert.Equal(t, tc.expectedResponse, resp)
+
+		})
+	}
+
+}
+
+func TestMlflowClient_SearchRunData(t *testing.T) {
+	tests := []struct {
+		name             string
+		idRun            string
+		expectedRespJSON string
+		expectedResponse SearchRunResponse
+		expectedError    error
+	}{
+		{
+			name:             "Valid Search",
+			idRun:            "abcdefg1234",
+			expectedRespJSON: RunSuccessJSON,
+			expectedResponse: SearchRunResponse{
+				RunData: RunResponse{
+					Info: RunInfo{
+						RunId:          "run-123",
+						ExperimentId:   "1",
+						UserId:         "root",
+						LifecycleStage: "active",
+						ArtifactURI:    "gs://my-bucket/run-123",
+					},
+					Data: RunData{
+						Tags: []RunTag{
+							{Key: "env", Value: "prod"},
+							{Key: "version", Value: "1.0.0"},
+						},
+					},
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:             "No related runs",
+			idRun:            "xytspow3412oi",
+			expectedRespJSON: `{}`,
+			expectedResponse: SearchRunResponse{},
+			expectedError:    nil,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+				w.WriteHeader(http.StatusOK)
+				_, err := w.Write([]byte(tc.expectedRespJSON))
+				require.NoError(t, err)
+			}))
+			defer server.Close()
+			client := NewMlflowClient(server.Client(), Config{
+				TrackingURL: server.URL,
+			}, &mocks.GcsService{})
+
+			resp, errAPI := client.SearchRunData(tc.idRun)
+
+			assert.Equal(t, tc.expectedError, errAPI)
+			assert.Equal(t, tc.expectedResponse, resp)
+		})
+	}
+}
+
+func TestMlflowClient_DeleteExperiment(t *testing.T) {
+	tests := []struct {
+		name             string
+		idExperiment     string
+		expectedRespJSON string
+		expectedError    error
+		httpStatus       int
+	}{
+		{
+			name:             "Valid Experiment Deletion",
+			idExperiment:     "1",
+			expectedRespJSON: `{}`,
+			expectedError:    nil,
+			httpStatus:       http.StatusOK,
+		},
+		{
+			name:             "ID not exist",
+			idExperiment:     "999",
+			expectedRespJSON: DeleteExperimentDoesntExist,
+			expectedError:    fmt.Errorf("No Experiment with id=999 exists"),
+			httpStatus:       http.StatusNotFound,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Just wondering, which HTTP endpoint operation is this mock server supposed to respond to? Is the `DeleteExperiment` function, multiple separate calls to the HTTP endpoint are made, one `POST` operation to minimally retrieve the list of runs and zero-to-many `POST` operations to actually delete the runs from the mlflow server. It seems like in this case the mocked server is hijacked to respond to all requests with the same mocked responses and errors 🤔